datadog_lambda/tracing.py

```diff
@@ -354,14 +356,16 @@ def extract_context_from_kinesis_event(event, lambda_context):
     return extract_context_from_lambda_context(lambda_context)
 
 
-def _deterministic_md5_hash(s: str) -> int:
-    """MD5 here is to generate trace_id, not for any encryption."""
-    hex_number = hashlib.md5(s.encode("ascii")).hexdigest()
-    binary = bin(int(hex_number, 16))
-    binary_str = str(binary)
-    binary_str_remove_0b = binary_str[2:].rjust(128, "0")
-    most_significant_64_bits_without_leading_1 = "0" + binary_str_remove_0b[1:-64]
-    result = int(most_significant_64_bits_without_leading_1, 2)
+def _deterministic_sha256_hash(s: str, part: str) -> (int, int):
+    sha256_hash = hashlib.sha256(s.encode()).hexdigest()
+
+    # First two chars is '0b'. zfill to ensure 256 bits, but we only care about the first 128 bits
+    binary_hash = bin(int(sha256_hash, 16))[2:].zfill(256)
+    if part == HIGHER_64_BITS:
+        updated_binary_hash = "0" + binary_hash[1:64]
+    else:
+        updated_binary_hash = "0" + binary_hash[65:128]
+    result = int(updated_binary_hash, 2)
```

Comment:
Here is an base 2 integer. But later, it is converted to a base 10 integer somewhere?

Reply:
I think we are returning in both binary and format.
- [returning binary](https://github.com/DataDog/datadog-lambda-python/blob/51b272dd04d250aa4aa1ede77b7ebf00365dab48/datadog_lambda/tracing.py#L386-L390)
- [convert into hex](https://github.com/DataDog/datadog-lambda-python/blob/51b272dd04d250aa4aa1ede77b7ebf00365dab48/datadog_lambda/tracing.py#L400-L402)